File: 0119-pascals-triangle-ii/0119-pascals-triangle-ii.ts
function getRow(rowIndex: number): number[] {
    const result = [[1]]

    for (let i = 0; i < rowIndex; i++) {
        const myRow = result[result.length - 1];
        const row = [1];

        for (let j = 1; j < myRow.length; j++) {
            row.push(myRow[j - 1] + myRow[j]);
        }
        row.push(1)
        result.push(row);
    }

    return result[rowIndex];
};

File: 0119-pascals-triangle-ii/0119-pascals-triangle-ii.py
class Solution:
    def getRow(self, rowIndex: int) -> List[int]:
        result = [[1]]
        for _ in range(rowIndex + 1):
            myRow = [0] + result[-1] + [0]
            row = []
            for i in range(len(result[-1]) + 1):
                row.append(myRow[i] + myRow[i + 1])
            result.append(row)
        
        return result[rowIndex]